src/strutil.c

```diff
@@ -0,0 +1,26 @@
+
+#include "ac_cfg.h"
+#include <string.h>
+#include "libavrdude.h"
+
+int str_starts(const char *str, const char *starts) {
+  return strncmp(str, starts, strlen(starts)) == 0;
+}
+
+int str_eq(const char *str1, const char *str2) {
+  return strcmp(str1, str2) == 0;
+}
+
+int str_contains(const char *str, const char *substr) {
+  return !!strstr(str, substr);
+}
+
+int str_ends(const char *str, const char *ends) {
+  size_t str_len  = strlen(str);
+  size_t ends_len = strlen(ends);
+
+  if (ends_len > str_len)
+    return 0;
+
+  return strncmp(str + str_len - ends_len, ends, ends_len) == 0;
```

Comment:
Works!

If you want it faster use `memcmp()` instead of `strncmp()`, if you want shorter source use `str_eq()` instead of `strncmp()` b/c (by construction) we *know* we have two strings of same size `ends_len` here.

Reply:
I'd actually prefer to use `str_eq`, since it means less code and is still easy to understand.